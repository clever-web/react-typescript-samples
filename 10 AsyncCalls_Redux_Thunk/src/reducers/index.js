var redux_1 = require('redux');
var memberReducer_1 = require('./memberReducer');
var membersReducer_1 = require('./membersReducer');
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = redux_1.combineReducers({
    member: memberReducer_1.default,
    members: membersReducer_1.default
});
//# sourceMappingURL=index.js.map