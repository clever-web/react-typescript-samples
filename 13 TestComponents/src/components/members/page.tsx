import * as React from 'react';
import { Link } from 'react-router-dom';
import { MemberEntity } from '../../model';
import { MemberHeader } from './memberHeader';
import { MemberRow } from './memberRow';

export interface Props {
  members: MemberEntity[];
  fetchMembers(): void;
}

export class MembersPage extends React.Component<Props,{}> {
  constructor(props) {
    super(props);
    this.state = { members: [] };
  }
  public componentDidMount() {
    this.props.fetchMembers();
  }

  public render() {
    return (
      <div className="row">
        <h2> Members Page</h2>
        <Link to="/member">New Member</Link>
        <table className="table">
          <thead>
            <MemberHeader />
          </thead>
          <tbody>
            {
              this.props.members.map((member) =>
                <MemberRow
                  key={member.id}
                  member={member}
                />
              )
            }
          </tbody>
        </table>
      </div>
    );
  }
};
