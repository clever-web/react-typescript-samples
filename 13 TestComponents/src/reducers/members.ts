import { actionTypes } from '../common/constants/actionTypes';
import { MemberEntity } from '../model';


export const createEmptyMembers = (): MemberEntity[] =>(
  []
)
export const membersReducer = (state: MemberEntity[] = [], action) => {
  switch (action.type) {
    case actionTypes.FETCH_MEMBERS_COMPLETED:
      return handleFetchMembersCompleted(state, action.payload);
  }

  return state;
};

const handleFetchMembersCompleted = (state: MemberEntity[], payload: MemberEntity[]) => {
  return payload;
};
