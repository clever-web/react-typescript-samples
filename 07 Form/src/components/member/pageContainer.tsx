import * as React from 'react';
import { hashHistory } from 'react-router';
import * as toastr from 'toastr';
import { memberAPI } from '../../api/member';
import { MemberEntity } from '../../model';
import { MemberPage } from './page';

interface State {
  member: MemberEntity;
}

export class MemberPageContainer extends React.Component<{}, State> {
  constructor() {
    super();

    this.state = {
      member: {
        id: -1,
        login: '',
        avatar_url: '',
      }
    };

    this.onFieldValueChange = this.onFieldValueChange.bind(this);
    this.onSave = this.onSave.bind(this);
  }

  private onFieldValueChange(fieldName: string, value: string) {
    const nextState = {
      ...this.state,
      member: {
        ...this.state.member,
        [fieldName]: value,
      }
    };

    this.setState(nextState);
  }

  private onSave() {
    memberAPI.saveMember(this.state.member)
      .then(() => {
        toastr.success('Member saved.');
        hashHistory.goBack();
      });
  }

  render() {
    return (
      <MemberPage
        member={this.state.member}
        onChange={this.onFieldValueChange}
        onSave={this.onSave}
      />
    );
  }
}
