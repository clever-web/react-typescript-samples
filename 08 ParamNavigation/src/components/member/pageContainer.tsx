import * as React from 'react';
import { hashHistory } from 'react-router';
import * as toastr from 'toastr';
import { memberAPI } from '../../api/member';
import { MemberEntity } from '../../model';
import { MemberPage } from './page';

interface Props {
  params: { id: string };
}

interface State {
  member: MemberEntity;
}

export class MemberPageContainer extends React.Component<Props, State> {
  constructor() {
    super();

    this.state = {
      member: {
        id: -1,
        login: '',
        avatar_url: '',
      }
    };

    this.onFieldValueChange = this.onFieldValueChange.bind(this);
    this.onSave = this.onSave.bind(this);
  }

  public componentDidMount() {
    const memberId = Number(this.props.params.id) || 0;
    memberAPI.fetchMemberById(memberId)
      .then((member) => {
        this.setState({
          ...this.state,
          member,
        });
      });
  }

  private onFieldValueChange(fieldName: string, value: string) {
    const nextState = {
      ...this.state,
      member: {
        ...this.state.member,
        [fieldName]: value,
      }
    };

    this.setState(nextState);
  }

  private onSave() {
    memberAPI.saveMember(this.state.member)
      .then(() => {
        toastr.success('Member saved.');
        hashHistory.goBack();
      });
  }

  render() {
    return (
      <MemberPage
        member={this.state.member}
        onChange={this.onFieldValueChange}
        onSave={this.onSave}
      />
    );
  }
}
