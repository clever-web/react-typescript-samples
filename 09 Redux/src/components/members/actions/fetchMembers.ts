import { actionTypes } from '../../../common/constants/actionTypes';
import { MemberEntity } from '../../../model';
import { memberAPI } from '../../../api/member';

export const fetchMembersAction = () => (dispatch) => {
  memberAPI.fetchMembers()
    .then((members) => {
      dispatch(fetchMembersCompleted(members));
    });
};

const fetchMembersCompleted = (members: MemberEntity[]) => ({
  type: actionTypes.FETCH_MEMBERS_COMPLETED,
  payload: members,
});
