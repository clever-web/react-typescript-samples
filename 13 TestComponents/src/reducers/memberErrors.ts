import { FieldValidationResult, FormValidationResult } from 'lc-form-validation';
import { actionTypes } from '../common/constants/actionTypes';
import { MemberErrors } from '../model';
import { MemberFieldChangePayload } from '../components/member/actions/memberFieldChange';

export const createEmptyMemberErrors = (): MemberErrors => ({
  login: new FieldValidationResult(),
});

export const memberErrorsReducer = (state = createEmptyMemberErrors(), action) => {
  switch (action.type) {
    case actionTypes.FETCH_MEMBERS_COMPLETED:
      return handleFetchMembersCompleted(state, action.payload);
    case actionTypes.UPDATE_MEMBER_FIELD:
      return handleUpdateMemberField(state, action.payload);
    case actionTypes.SAVE_MEMBER:
      return handleSaveMember(state, action.payload);
  }

  return state;
};

const handleFetchMembersCompleted = (state: MemberErrors, payload) => {
  return createEmptyMemberErrors();
};

const handleUpdateMemberField = (state: MemberErrors, payload: MemberFieldChangePayload): MemberErrors => {
  return {
    ...state,
    [payload.fieldValidationResult.key]: payload.fieldValidationResult,
  };
};

const handleSaveMember = (state: MemberErrors, payload: FormValidationResult): MemberErrors => {
  const newMemberErrors = { ...state };

  return payload.fieldErrors.reduce((memberErrors, fieldValidationResult) => {
    memberErrors[fieldValidationResult.key] = fieldValidationResult;
    return memberErrors;
  }, newMemberErrors);
};
