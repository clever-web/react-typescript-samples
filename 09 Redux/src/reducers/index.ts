import { combineReducers } from 'redux';
import { MemberEntity, MemberErrors } from '../model';
import { membersReducer } from './members';
import { memberReducer } from './member';
import { memberErrorsReducer } from './memberErrors';

export interface State {
  members: MemberEntity[];
  member: MemberEntity;
  memberErrors: MemberErrors;
};

export const state = combineReducers<State>({
  members: membersReducer,
  member: memberReducer,
  memberErrors: memberErrorsReducer,
});
