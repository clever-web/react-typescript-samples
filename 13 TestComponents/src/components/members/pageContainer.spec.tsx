import * as React from 'react';
import { shallow } from 'enzyme';
import { State } from '../../reducers';
import configureStore from 'redux-mock-store';
import { createEmptyMembers } from '../../reducers/members';
import { createEmptyMember } from '../../reducers/member';
import { createEmptyMemberErrors } from '../../reducers/memberErrors';
import { MembersPageContainer } from './pageContainer';
import * as FetchMembers from './actions/fetchMembers';

const getMockStore = configureStore();

describe('/components/members/pageContainer', () => {
    it('should render as spected passing state',()=>{
        //arrange
        const state = {
            members: createEmptyMembers(),
            member: createEmptyMember(),
            memberErrors: createEmptyMemberErrors()
        } as State;
        const store = getMockStore(state);
        //act
        
        const component =  shallow(
            <MembersPageContainer/>,
            {
                context: {store},
            },
        );
        //assert
        expect(component).toMatchSnapshot();
    });
    it('should call to fetchMembers action creator when call to fetchMembers prop', () => {
        // Arrange
        const state = {
            members: createEmptyMembers(),
            member: createEmptyMember(),
            memberErrors: createEmptyMemberErrors()
        } as State;


        const store = getMockStore(state);
        const fetchMembersStub = jest.spyOn(FetchMembers, 'fetchMembersAction')
            .mockImplementation(() => ({
                type: 'test action type',
            }));

        // Act
        const component = shallow(
            <MembersPageContainer
            />,
            {
                context: { store },
            },
        );

       
        // Assert
        expect(fetchMembersStub).toMatchSnapshot();
    }); 
});