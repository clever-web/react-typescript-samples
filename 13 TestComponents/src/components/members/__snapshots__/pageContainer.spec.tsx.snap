// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/members/pageContainer should call to fetchMembers action creator when call to fetchMembers prop 1`] = `[MockFunction]`;

exports[`/components/members/pageContainer should render as spected passing state 1`] = `
<MembersPage
  fetchMembers={[Function]}
  members={Array []}
/>
`;
